pkgs/development/interpreters/python/build-python-package-pyproject.nix

```diff
@@ -0,0 +1,37 @@
+# This function provides specific bits for building a setuptools-based Python package.
+
+{ lib
+, python
+}:
+
+{
+# passed to "python setup.py build_ext"
+# https://github.com/pypa/pip/issues/881
+  setupPyBuildFlags ? []
+# Execute before shell hook
+, preShellHook ? ""
+# Execute after shell hook
+, postShellHook ? ""
+, ... } @ attrs:
+
+attrs // {
+  # we copy nix_run_setup over so it's executed relative to the root of the source
+  # many project make that assumption
+  buildPhase = attrs.buildPhase or ''
+    runHook preBuild
+    mkdir -p dist
+    echo "Creating a wheel..."
+    ${python.pythonForBuild.interpreter} -m pip wheel --no-index --no-deps --no-clean --no-build-isolation --wheel-dir dist .
+    echo "Finished creating a wheel..."
+    runHook postBuild
+  '';
+
+  doCheck = attrs.doCheck or true;
+
+  installCheckPhase = attrs.checkPhase or ''
+    runHook preCheck
+    ${python.interpreter} -m unittest discover
```

Comment:
Perhaps better to exit with a non-zero status and require a `checkPhase` to be passed in.